libr/core/linux_heap_glibc.c

```diff
@@ -21,6 +21,32 @@
 #define GHT_MAX UT64_MAX
 #endif
 
+
+static GHT GH(get_va_symbol)(RCore *core, const char *path, const char *symname) {
+	GHT vaddr = GHT_MAX;
+	RBin *bin = core->bin;
+	RBinFile *current_bf = r_bin_cur (bin);
+	RListIter *iter;
+	RBinSymbol *s;
+
+	RBinOptions opt;
+	r_bin_options_init (&opt, -1, 0, 0, false);
+	r_bin_open (bin, path, &opt);
+	RList *syms = r_bin_get_symbols (bin);
+
+	r_list_foreach (syms, iter, s) {
+		if (!strcmp (s->name, symname)) {
+			vaddr = s->vaddr;
+			break;
+		}
+	}
+
+	RBinFile *libc_bf = r_bin_cur (bin);
+	r_bin_file_delete (bin, libc_bf->id);
+	r_bin_file_set_cur_binfile (bin, current_bf);
```

Comment:
this is another reason why cur must die. current-bf can be invalidated at any time. so using "safe" fd style identifiers would make that more error-prone and there wont be a need to tell r2 which is the current bin loaded, because the selected binfile can be held by RCore or just be selected depènding on the offset like its done by r2frida or when `e bin.at` is set